adapters/commons/src/main/java/io/vanillabp/cockpit/adapter/common/usertask/kafka/UserTaskKafkaPublishing.java

```diff
@@ -0,0 +1,175 @@
+package io.vanillabp.cockpit.adapter.common.usertask.kafka;
+
+import io.vanillabp.cockpit.adapter.common.CockpitProperties;
+import io.vanillabp.cockpit.adapter.common.usertask.UserTaskProperties;
+import io.vanillabp.cockpit.adapter.common.usertask.UserTaskPublishing;
+import io.vanillabp.cockpit.adapter.common.usertask.UserTasksWorkflowProperties;
+import io.vanillabp.cockpit.adapter.common.usertask.events.UserTaskActivatedEvent;
+import io.vanillabp.cockpit.adapter.common.usertask.events.UserTaskCancelledEvent;
+import io.vanillabp.cockpit.adapter.common.usertask.events.UserTaskCompletedEvent;
+import io.vanillabp.cockpit.adapter.common.usertask.events.UserTaskCreatedEvent;
+import io.vanillabp.cockpit.adapter.common.usertask.events.UserTaskEvent;
+import io.vanillabp.cockpit.adapter.common.usertask.events.UserTaskSuspendedEvent;
+import io.vanillabp.cockpit.adapter.common.usertask.events.UserTaskUiUriType;
+import io.vanillabp.cockpit.adapter.common.usertask.events.UserTaskUpdatedEvent;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.kafka.core.KafkaTemplate;
+import org.springframework.util.StringUtils;
+
+import java.util.List;
+
+public class UserTaskKafkaPublishing implements UserTaskPublishing {
+
+    private static final Logger logger = LoggerFactory.getLogger(UserTaskKafkaPublishing.class);
+
+    private final String workerId;
+
+    private final UserTaskProtobufMapper userTaskMapper;
+
+    private final CockpitProperties properties;
+
+    private final UserTasksWorkflowProperties workflowsCockpitProperties;
+
+    private final KafkaTemplate<String, byte[]> kafkaTemplate;
+
+    private static final UserTaskProperties EMPTY_PROPERTIES = new UserTaskProperties();
+
+
+    public UserTaskKafkaPublishing(
+            String workerId,
+            CockpitProperties properties,
+            UserTasksWorkflowProperties workflowsCockpitProperties,
+            UserTaskProtobufMapper userTaskMapper,
+            KafkaTemplate<String, byte[]> kafkaTemplate
+    ) {
+        this.workerId = workerId;
+        this.userTaskMapper = userTaskMapper;
+        this.properties = properties;
+        this.workflowsCockpitProperties = workflowsCockpitProperties;
+        this.kafkaTemplate = kafkaTemplate;
+    }
+
+    @Override
+    public void publish(String apiVersion, List<UserTaskEvent> events) {
+
+        if (apiVersion.equals("v1")) {
```

Comment:
I think we should remove "apiVersion" and "ApiVersionAware". I guess this is an overkill since the version could be hard-coded in the publisher:
1. The version used for REST is implicitly defined by the REST client classes generated and has to match the REST endpoint configured. This could be tested on startup by testing the URL containing the hard-coded version identifier ("/v1/").
2. The version used for Kafka should be hard-coded and sent as a Kafka-header

The rest of the story has to be implemented by business-cockpit which consumes those events:
1. All versions of REST endpoints have to be implemented
2. All Kafka-events have to be consumed and upcasted according to the version given in the event's header